### DISCUSSION TITLE
How can i search data with using "doc_id < 5"???

### DISCUSSION THREAD

**Discussion by Asidistaken** [msiemens/tinydb#549] [Q&A]
How can i search data with using "doc_id < 5"???

**Comment by n3storm** [comment]
What is the code you are working with?

**Comment by TheNewThinkTank** [comment]
Here I am assuming that "doc_id" is a field in your data, and not the database table's id's. based on the official docs: https://tinydb.readthedocs.io/en/latest/usage.html#queries
```Python
from tinydb import TinyDB, Query

# Create or load the TinyDB database
db = TinyDB('db.json')

# Define the Query object
query = Query()

# Search for data where doc_id is less than 5
result = db.search(query.doc_id < 5)

# Print the result
print(result)
```

**Reply by msiemens** [reply]
Actually, this doesn't work because this will try to get a field named `doc_id` from the document entry, instead of testing the document's ID which is not part of the `dict` data.

**Comment by msiemens** [comment]
This is not possible with TinyDB's query mechanism because it only works with the documents' data, NOT their metadata (i.e. the document ID). I think the most straightforward solution here is something like this:

```python
[d for d in db.all() if d.doc_id < 5]
```

Using `db.search` basically does the same thing internally (except for converting the data to a `Document` instance after filtering), so there is no real downside except that this would not be stored in the query cache (which I think in most cases won't really be an issue).
